srcs/parsing/tokens.c

```diff
@@ -88,35 +30,24 @@ static char	**fill_tokens(const char *s, char c, int words, char **tokens)
 			i++;
 		letters = 0;
 		while (s[i + letters] && s[i + letters] != c)
-		// here check to get the righ number of letters for the quote
-			letters++;
-		tokens[word] = (char *)malloc(sizeof(char) * (letters + 1));
-		if (!tokens[word])
-			return (NULL);
-		letters = 0;
-		while (s[i] && s[i] != c)
-			tokens[word][letters++] = s[i++];
-		tokens[word++][letters] = 0;
+			letters = handle_quotes(s, i, letters, &tokens[word], c) + 1; // here we do letters++ every time + length of quote if there's one to handle.
```

Comment:
handle_quotes doesn't tell me what the function actually does, could we change it so that it's clearer that it returns the number of letters in a string?